lib/daru/io/io.rb

```diff
@@ -201,25 +199,29 @@ def from_csv_prepare_opts opts
 
       def from_csv_hash_with_headers(path, opts)
         opts[:header_converters] ||= :symbol
-
         ::CSV
-          .read(path, 'rb',opts)
+          .parse(new_open(path), opts)
           .tap { |c| yield c if block_given? }
           .by_col.map { |col_name, values| [col_name, values] }.to_h
       end
 
       def from_csv_hash(path, opts)
         csv_as_arrays =
           ::CSV
-          .open(path, 'rb', opts)
+          .parse(new_open(path), opts)
           .tap { |c| yield c if block_given? }
           .to_a
-
         headers       = ArrayHelper.recode_repeated(csv_as_arrays.shift)
         csv_as_arrays = csv_as_arrays.transpose
-
         headers.each_with_index.map { |h, i| [h, csv_as_arrays[i]] }.to_h
       end
+
+      def new_open(url)
+        open(url)
+      rescue URI::InvalidURIError
+        host = url.match(".+\:\/\/([^\/]+)")[1]
+        Net::HTTP.get host, url.partition(host)[2] || '/'
```

Comment:
Well, this mindset is praiseworthy, yet it can easily lead us into integrating wrong data encoding guessers, redirect handlers and generic `Daru.find(data_about: 'pretty kittens')` things. So we do YAGNI when we can, unless this aproach is proved to be wrong (for this particular problem: IF we'll know there is significant number of daru users on Ruby 2.1 AND significant number of their data on cool_underscored_domains, we could rethink our approach).

Currently, it looks for me as small, yet existing complication with very low value for users. (BTW, even if implementing this; for this kind of things, like "fix the bug for particular Ruby version", I'd prefer just core ext with `if RUBY_VERSION < '2.2'` guard).